lib/active_resource/base.rb

```diff
@@ -652,11 +652,11 @@ def connection(refresh = false)
       end
 
       def headers
-        self._headers ||= {}
-        if superclass != Object && superclass.headers
-          self._headers = superclass.headers.merge(_headers)
+        headers_state = self._headers || {}
+        if superclass != Object
```

Comment:
`superclass.headers` will return a `{}` even if it's value is not set, so it will always evaluate to `true`. Given this we can safely remove it from the conditional. @rafaelfranca 👍